src/hotspot/share/oops/instanceKlass.cpp

```diff
@@ -3859,9 +3860,15 @@ void InstanceKlass::print_class_load_cause_logging() const {
       // Log to string first so that lines can be indented
       stringStream stack_stream;
       char buf[O_BUFLEN];
-      frame f = os::current_frame();
-      VMError::print_native_stack(&stack_stream, f, current, true /*print_source_info */,
-                                  -1 /* max stack_stream */, buf, O_BUFLEN);
+      address lastpc = nullptr;
+      if (os::platform_print_native_stack(&stack_stream, nullptr, buf, O_BUFLEN, lastpc)) {
+        // We have printed the native stack in platform-specific code,
+        // so nothing else to do in this case.
+      } else {
```

Comment:
Why did you do this? Did you observe problems on Windows? This is too low-level for this code. If we need to do this then it needs to be pushed down to a lower-level and encapsulated more cleanly. We do not do this in `JavaThread::print_jni_stack()`.

I need to revoke my approval while this is resolved.

Reply:
I'm also investigating the details of the stack printing code - use of `os::platform_print_native_stack` is very inconsistent.

Reply:
It seems it is. I was not aware of that. So I need to fix `JavaThread::print_jni_stack`, and more generally consolidate this stack printing code.

Reply:
Yes I will do the cleanup in a follow up issue. I will approve this PR now. You still need a second reviewer to approve.

Thanks.